### ISSUE TITLE
Preserve numeric keys in data/file merge in ContentValidationListener::onRoute()

### ISSUE THREAD

**Issue by Jakovitz** [zfcampus/zf-content-validation#62]
ContentValidationListener::onRoute() uses array_merge_recursive() which doesn't preserve numeric keys.

```php
$data[0]['foo'] = 'bar';
$files[0]['file'] = [
    'name' => 'foo.txt',
    'type' => 'text/plain',
    'size' => 1,
    'tmp_name' => '/tmp/foo.txt',
    'error' => UPLOAD_ERR_OK,
];
```

will be merged into

```php
[
    0 => [
        'foo' => 'bar',
    ],
    1 => [
        'file' => [
            'name' => 'foo.txt',
            'type' => 'text/plain',
            'size' => 1,
            'tmp_name' => '/tmp/foo.txt',
            'error' => UPLOAD_ERR_OK,
        ],
    ],
];
```

Is this intentional? I would expect it to be

```php
[
    0 => [
        'foo' => 'bar',
        'file' => [
            'name' => 'foo.txt',
            'type' => 'text/plain',
            'size' => 1,
            'tmp_name' => '/tmp/foo.txt',
            'error' => UPLOAD_ERR_OK,
        ],
    ],
];
```

**Comment by Jakovitz** [comment]
I'm wondering if this is getting looked at? Currently, the only way to circumvent this, is to extend ZF\ContentValidation\ContentValidationListener, overwrite the entire onRoute() function, and create a new factory, just to change a single line in onRoute().

**Comment by webimpress** [comment]
Resolved in #63, released [1.2.0](https://github.com/zfcampus/zf-content-validation/releases/tag/1.2.0). Can be closed.
